common/appointment/create.ts

```diff
@@ -0,0 +1,82 @@
+import { Field, InputType, Int } from 'type-graphql';
+import { prisma } from '../prisma';
+import { getUserIdTypeORM } from '../user';
+import assert from 'assert';
+import { getUserForTypeORM } from '../user';
+import { lecture_appointmenttype_enum } from '../../graphql/generated';
+
+@InputType()
+export abstract class AppointmentCreateInputBase {
+    @Field({ nullable: true })
+    title?: string;
+    @Field({ nullable: true })
+    description?: string;
+    @Field()
+    start: Date;
+    @Field()
+    duration: number;
+    @Field({ nullable: true })
+    meetingLink?: string;
+}
+
+@InputType()
+export abstract class AppointmentCreateMatchInput extends AppointmentCreateInputBase {
+    @Field(() => Int)
+    matchId: number;
+    @Field(() => lecture_appointmenttype_enum)
+    appointmentType: 'match';
+}
+
+@InputType()
+export abstract class AppointmentCreateGroupInput extends AppointmentCreateInputBase {
+    @Field(() => Int, { nullable: false })
+    subcourseId: number;
+    @Field(() => lecture_appointmenttype_enum)
+    appointmentType: 'group';
+}
+
+export const createMatchAppointments = async (matchId: number, appointmentsToBeCreated: AppointmentCreateMatchInput[]) => {
+    const { pupil, student } = await prisma.match.findUniqueOrThrow({ where: { id: matchId }, include: { student: true, pupil: true } });
+    const studentUserId = getUserIdTypeORM(student);
+    const pupilUserId = getUserIdTypeORM(pupil);
+    return await Promise.all(
+        appointmentsToBeCreated.map(
+            async (appointmentToBeCreated) =>
+                await prisma.lecture.create({
```

Comment:
Maybe createMany?

Reply:
@LomyW could you bring this change in, in the integration test branch? Would be a charm 🥇